crates/holochain_types/src/app/app_manifest/app_manifest_v1.rs

```diff
@@ -194,6 +194,44 @@ impl AppManifestV1 {
         }
     }
 
+    /// Selectively overrides the modifiers for the given roles. Only fields with value `Some(T)` will
+    /// override the corresponding value in the manifest. If `None` is provided for a modifier field
+    /// the corresponding value in the manifest will remain untouched.
+    pub fn override_modifiers(
+        &mut self,
+        modifiers: HashMap<RoleName, DnaModifiersOpt<YamlProperties>>,
+    ) -> AppManifestResult<()> {
+        let existing_role_names = self
+            .roles
+            .iter()
+            .map(|manifest| &manifest.name)
+            .collect::<Vec<&String>>();
+        for role_name in modifiers.keys() {
+            if !existing_role_names.contains(&role_name) {
+                return Err(AppManifestError::InvalidRoleName(format!(
+                    "Tried to set modifiers for a role name that does not exist in the dna manifest: {role_name}"
+                )));
+            }
+        }
+        for role in self.roles.iter_mut() {
```

Comment:
This seems reasonable, to just modify the manifest. However, it introduces something that couldn't happen before. We would previously have been able to calculate the DNA hash just from the manifest. When we come to do coordinator updates, whether with the current API, or the new one when we get to it - we will end up rejecting the incoming manifest because the DNA hash won't match unless these override properties are provided. 

Should the overrides be stored alongside the app state in the conductor state so that this can be recalculated later against an updated manifest?

Reply:
Hmm...the network seed was already [overwritten as well](https://github.com/holochain/holochain/blob/84f73707319dff3f0feab4918e63f173766f9f83/crates/holochain/src/conductor/conductor.rs#L1686) so this problem must have been there before? I just chose the same approach as was used with the network seed...

I'm also not sure what you mean by
> We would previously have been able to calculate the DNA hash just from the manifest

What do you mean by "just from the manifest"? Where would that non-overriden manifest be taken from?

Reply:
Yes, the problem was already there then. What should we do about it from here? I think this is something that could easily be missed when we come to implement the coordinator updates, so how can we help ourselves out without doing loads of extra work now?

I'd suggest either leaving a code comment explaining the impact of this block of code that can update either the network seed or do role-specific overrides. Or instead, updating the coordinator update spec to mention this.

Reply:
As discussed in another channel, we should not need to care about the dna hash when updating coordinator zomes, the integrity zome wasm hashes should be sufficient for determining compatibility, i.e. independent of modifiers.